application/common/models/Reset.php

```diff
@@ -43,14 +60,159 @@ public function rules()
+            }
+        }
+
+        /*
+         * Send Reset to appropriate method
+         */
+        $reset->send();
+
+        return $reset;
+    }
+
+    /**
+     * Send reset notification to appropriate method
+     * @throws \Exception
+     */
+    public function send()
+    {
+        /*
+         * Based on type/method send reset verification and update
+         * model with reset code
+         */
+        switch ($this->type) {
+            case self::TYPE_PRIMARY:
+                $this->sendPrimary();
+                break;
+            case self::TYPE_SUPERVISOR:
+                $this->sendSupervisor();
+                break;
+            case self::TYPE_SPOUSE:
+                $this->sendSpouse();
+                break;
+            case self::TYPE_METHOD:
+                $this->sendMethod();
+                break;
+            default:
+                throw new \Exception('Reset is configured with unknown type.', 1456784825);
+        }
+    }
+
+    public function sendPrimary()
+    {
+        /**
+         * @todo send email to $user->email with reset code
+         */
+    }
+
+    public function sendSupervisor()
+    {
+        /**
+         * @todo if $this->user->hasSupervisor(), send reset
+         *       code to $this->user->getSupervisorEmail()
+         */
+    }
+
+    public function sendSpouse()
+    {
+        /**
+         * @todo if $this->user->hasSpouse(), send reset
+         *       code to $this->user->getSpouseEmail()
+         */
+    }
+
+    /**
+     * Determine if type is phone or email and send accordingly
+     * @throws \Exception
+     */
+    public function sendMethod()
+    {
+        if ( ! ($this->method instanceof Method)) {
+            throw new \Exception('Method not initialized on Reset', 1456608512);
+        }
+
+        if ($this->method->type == Method::TYPE_EMAIL) {
+            $this->sendEmail();
+        } elseif ($this->method->type == Method::TYPE_PHONE) {
+            $this->sendPhone();
+        } else {
+            throw new \Exception('Method using unknown type', 1456608781);
+        }
+    }
+
+    public function sendEmail()
+    {
+        /**
+         * @todo send email to $this->method->value with reset code
+         */
+    }
+
+    public function sendPhone()
+    {
+        /**
+         * @todo use phone service component to send notification to $this->method->value
+         */
+    }
+
+    /**
      * Calculate expiration timestamp based on given timestamp and configured reset lifetime
-     * @param null $time
+     * @param null|integer $time
      * @return integer
      */
-    public static function getExpireTimestamp($time=null)
+    public static function getExpireTimestamp($time = null)
     {
         $time = is_null($time) ? time() : $time;
 
-        return $time+\Yii::$app->params['reset']['lifetimeSeconds'];
+        return $time + \Yii::$app->params['reset']['lifetimeSeconds'];
```

Comment:
Would it be worth adding a sanity check here to make sure the resulting value isn't too far in the past or the future (or at least ensuring that the result is an integer)? Just a thought.